### ISSUE TITLE
Clang cannot compile ttmath Uint in a seq

### ISSUE THREAD

**Issue by mratsim** [status-im/nim-ttmath#10]
See https://github.com/status-im/nimbus/issues/2.

This is needed for nimbus, workaround is using GCC on MacOS but probably we cant use LLVM at all for an eventual WASM target.

```Nim
import ../src/ttmath

type
  Stack = object
    data: seq[Uint256]

proc newStack(): Stack =
  result.data = @[]
```

```
ttmath_stack.cpp:61:29: error: flexible array member 'data' of type 'TY_fSgjjjotfhO8wSAxXj6bfw []' with non-trivial destruction
  TY_fSgjjjotfhO8wSAxXj6bfw data[SEQ_DECL_SIZE];
```

```C++
// Line 60-62
struct tySequence_sp47cI5UpQ9bwXDfsT164pg : TGenericSeq {
  TY_fSgjjjotfhO8wSAxXj6bfw data[SEQ_DECL_SIZE];
};

// line 138-145
N_LIB_PRIVATE N_NIMCALL(void, newStack_k6WSwonLTc7xJzVEEj9c9a1Q)(tyObject_Stack_9cQNCXzCX5qwQ4cPhPCufuA* Result) {
	nimfr_("newStack", "stack.nim");
	chckNil((void*)Result);
	genericReset((void*)Result, (&NTI_9cQNCXzCX5qwQ4cPhPCufuA_));
	nimln_(9, "stack.nim");
	unsureAsgnRef((void**) (&(*Result).data), (tySequence_sp47cI5UpQ9bwXDfsT164pg*) newSeq((&NTI_sp47cI5UpQ9bwXDfsT164pg_), 0));
	popFrame();
}
```

**Comment by mratsim** [comment]
-Wflexible-array-extensions does not seem to work.

**Comment by mratsim** [comment]
Fixed by #15
